### PR TITLE
Feature/freight status broadcast

### LINKED ISSUE #824: Port drayage plugin should broadcast status message when stopped
### Types of Issue

- [ ] Anomaly report (something appears to not work correctly)
- [x] Enhancement request (describe the enhancement being requested)
- [ ] Other (please ensure the description clarifies why the issue doesn’t fall into either of the above categories)

### Descriptive summary
In order to support CARMA freight demonstration needs the CARMA platform needs a CARMA Port Drayage Plugin. One feature of this port drayage plugin should be to broadcast it's status message (in the form of a mobility operations message) once it has commanded the vehicle to a stop at the various stages of the Port Drayage process. This message should contain the standard CARMA vehicle identifiers as well as a specific CMV ID and container ID.

### Expected behavior
1. If the port drayage plugin detects a vehicle stop maneuver planned by itself it should broadcast the status message.

### Software version this applies to
develop

### Actual behavior
N/a, not yet implemented
### Steps to reproduce the behavior

N/a

### Related work

N/a

### PR DESCRIPTION
**Pull request by kjrush**
# PR Details
## Description

Implementation of the initial skeleton for the Port Drayage Plugin, with the initial functionality of broadcasting the status message once it has commanded a stop (though the stopping functionality has not yet been implemented yet, and will be completed under a separate set of stories).

## Related Issue

Closes #824 
## Motivation and Context

This change resolves issue #824 by broadcasting a message when a stop maneuver planned by the port drayage plugin has been completed. 

## How Has This Been Tested?

This has not yet been unit tested, I am working through that currently but wanted to get this code up for review prior to that to expedite the process.

## Types of changes

- [ ] Defect fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that cause existing functionality to change)

## Checklist:

- [x] I have added any new packages to the sonar-scanner.properties file
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
[CARMA Contributing Guide](https://github.com/usdot-fhwa-stol/carma-platform/blob/develop/Contributing.md) 
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: port_drayage_plugin/src/port_drayage_plugin.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            speed_epsilon
+        };
+        
+        ros::Subscriber maneuver_sub = _nh->subscribe<cav_msgs::ManeuverPlan>("final_Maneuver_plan", 5, 
+            [&](const cav_msgs::ManeuverPlanConstPtr& plan) {
+                pdw.set_maneuver_plan(plan);
+        });
+        _maneuver_plan_subscriber = std::make_shared<ros::Subscriber>(maneuver_sub);
+
+        ros::Subscriber twist_sub = _nh->subscribe<geometry_msgs::TwistStamped>("speed", 5,
```
Comment (saina-ramyar) [inline port_drayage_plugin/src/port_drayage_plugin.cpp:53]: have you checked the topic name speed? The hardware_interface/can/speed topic I saw is a standard float not twist.

File: port_drayage_plugin/src/port_drayage_worker.cpp [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                        return true;
+                    }
+                }
+            }
+        }
+
+        return false;
+    }
+
+    void PortDrayageWorker::spin() {
```
Comment (saina-ramyar) [inline port_drayage_plugin/src/port_drayage_worker.cpp:45]: is this called anywhere?

File: port_drayage_plugin/src/port_drayage_worker.cpp [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+        if (plan == nullptr || speed == nullptr) {
+            ROS_WARN("Checking for stop when PortDrayagePlugin not properly initialized. Speed or plan is null");
+            return false;
+        }
+
+        if (plan->maneuvers.size() > 0) {
+            if (plan->maneuvers[0].type == cav_msgs::Maneuver::STOP_AND_WAIT) {
+                cav_msgs::Maneuver stop_maneuver = _cur_plan->maneuvers[0];
+                if (stop_maneuver.stop_and_wait_maneuver.parameters.planning_strategic_plugin == PORT_DRAYAGE_PLUGIN_ID) {
+                    double longitudinal_speed = speed->twist.linear.x; // TODO: Verify that X is longitudinal speed of vehicle
```
Comment (saina-ramyar) [inline port_drayage_plugin/src/port_drayage_worker.cpp:34]: yes, twist.linear.x is the longitudinal speed of the vehicle. Please make sure the "speed" is twist type

**Comment by kjrush** [comment]
Merging with @saina-ramyar 's approval since there still appears to be an issue with the SONAR_STOP tags for coverage reporting.

**Pull request merged**